File: ForumMCBackend/ForumMCBackend/Models/Account.cs
using ForumMCBackend.Db;
using System.Security.Claims;

namespace ForumMCBackend.Models
{
    public enum AccountRoles
    {
        USER,
        MODERATOR,
        ADMIN,
    }

    public class Account : BaseEntity
    {
        public string? UserName { get; set; }
        public string? Password { get; set; }
        public AccountRoles Role { get; set; } = AccountRoles.USER;
    
        static public bool IsInRoles(ClaimsPrincipal user, List<AccountRoles> roles)
        {
            foreach(var role in roles)
            {
                if(user.IsInRole(role.ToString()))
                {
                    return true;
                }
            }

            return false;            
        }

        static public bool IsInRole(ClaimsPrincipal user, AccountRoles role)
        {
            return user.IsInRole(role.ToString());
        }
    }
}

File: forummc-frontend/src/models/account.ts
interface Account {
    id: number,
    userName: string | null,
    role: AccountRoles,
    createdAt: number,
    updatedAt: number,
}

enum AccountRoles {
    User,
    Moderator,
    Admin,
}

interface Message {

    id: number,
    bodyText: string,
    account: Account,
    goesLive: number,
    isHidden: boolean,
    inReplyTo: Message,
    createdAt: number,
    updatedAt: number,
}

interface Topic {

    id: number,
    createdBy: Account,
    category: Category,
    isHidden: boolean,
    createdAt: number,
    updatedAt: number,
}

interface Category {
    id: number,
    title: string,
    createdAt: number,
    updatedAt: number,
}

